sfge/src/main/java/com/salesforce/rules/PostProcessingRule.java

```diff
@@ -0,0 +1,22 @@
+package com.salesforce.rules;
+
+import java.util.List;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+
+/**
+ * Interface that can be optionally implemented by subclasses of {@link AbstractPathBasedRule}. The
+ * {@link #postProcess} hook will be invoked during rule execution after all path entry-points have
+ * been fully evaluated. This allows the rule to gather information across all its path executions
+ * and then process that information at the very end. <br>
+ * An example of this interface in action is {@link UnusedMethodRule}, whose {@code run()} method
+ * notes every unique method invoked in any path, and whose {@link #postProcess} method creates
+ * violations for methods that were never invoked.
+ */
+public interface PostProcessingRule {
```

Comment:
@rmohan20 , what do you think of the names for this interface and its method? I'm totally open to changing them.

Reply:
@jfeingold35 this interface and method, they aren't public-facing? Not exposed to end users e.g. not UI text?

Reply:
@jfeingold35 Or are we saying that we have a net new rule, PostProcessingRule, that combines public and protected instance methods and ... does what? Let's collaborate in the UI text doc?

Reply:
@teresa-allen-sfdc This would be internal only. We have ways to group rules so that we know which parts to execute and when. The grouping is more operational than functional.

@jfeingold35 "PostProcessingRule" makes sense. How about "InformationGatheringRule" or "DataCollectionAndProcessingRule"? We can replace the `_run(...)` with a `_collect(...)` method so that we don't have to return back an empty list of violations.